tools/tplg_parser/tokens.c

```diff
@@ -136,8 +136,8 @@ int sof_parse_token_sets(void *object, const struct sof_topology_token *tokens,
 
 		array = MOVE_POINTER_BY_BYTES(array, array->size);
 
-		if (found >= count) {
-			total += found;
+		if (found > 0) {
```

Comment:
@singalsu why should we care about found at all then? We should just get rid of it altogether no?

Reply:
Then none of the parsed tokens would be applied (or like now apply them to 1st format entry only). Please try topologies with multiple format PCMs with SOF plugin to see if you have this issue. I'm running topologies sof-hda-benchmark-comp/wordlength.tplg:

![sof-hda-benchmark-eqiir32](https://github.com/user-attachments/assets/ab49b09f-652b-4df5-8af3-5c8e98bc6179)

Reply:
Note: I don't use those HDMI pipelines but there's no simple way to get rid of them in these test toplogies based on hda-generic. I can run these in both a real HDA device and with testbench to run both capture and playback (use -p 1,2,3,4 or - p 1,2 for playback or -p 3,4 for capture as currently there is no automatic discovery for pipelines to enable).